scene/main/node.cpp

```diff
@@ -1685,13 +1685,21 @@ Node *Node::get_child(int p_index, bool p_include_internal) const {
 		if (p_index < 0) {
 			p_index += data.children_cache.size();
 		}
-		ERR_FAIL_INDEX_V(p_index, (int)data.children_cache.size(), nullptr);
+		ERR_FAIL_INDEX_V_MSG(p_index, (int)data.children_cache.size(), nullptr,
+				vformat("%s: Can't get child node at index %d, since the node only has %d children (including internal children).",
+						is_inside_tree() ? String(get_path()) : String(get_name()) + " (out-of-tree)",
+						p_index,
+						(int)data.children_cache.size()));
```

Comment:
Note that this would benefit from `Node.get_debug_path()`, which I added in https://github.com/godotengine/godot/pull/95579. This way, we can reduce code repetition significantly, as a lot of messages will eventually benefit from printing out node paths (and falling back to the node name if out-of-tree).